From 8d9c84dc28a9c280c9259b70f570b6a8410bba82 Mon Sep 17 00:00:00 2001
From: "K.Takata" <kentkt@csc.jp>
Date: Mon, 1 Oct 2018 06:51:20 +0900
Subject: [PATCH 03/27] nsis: Skip if libgcc_s_sjlj-1.dll doesn't exist

Next release of mlocati/gettext-iconv-windows might remove the
dependency on the DLL.
---
 nsis/gvim.nsi | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/nsis/gvim.nsi b/nsis/gvim.nsi
index 0a14a1f77..d8b54b613 100644
--- a/nsis/gvim.nsi
+++ b/nsis/gvim.nsi
@@ -380,7 +380,10 @@ Section "Add an Edit-with-Vim context menu entry"
 !ifdef HAVE_NLS
 	File ${GETTEXT}\gettext32\libintl-8.dll
 	File ${GETTEXT}\gettext32\libiconv-2.dll
+  !if /FileExists "${GETTEXT}\gettext32\libgcc_s_sjlj-1.dll"
+	# Install libgcc_s_sjlj-1.dll only if it is needed.
 	File ${GETTEXT}\gettext32\libgcc_s_sjlj-1.dll
+  !endif
 !endif
 
 	${If} ${Errors}
@@ -396,9 +399,12 @@ Section "Add an Edit-with-Vim context menu entry"
 	  GetTempFileName $3 $0\GvimExt32
 	  File /oname=$3 ${GETTEXT}\gettext32\libiconv-2.dll
 	  Rename /REBOOTOK $3 $0\GvimExt32\libiconv-2.dll
+  !if /FileExists "${GETTEXT}\gettext32\libgcc_s_sjlj-1.dll"
+	  # Install libgcc_s_sjlj-1.dll only if it is needed.
 	  GetTempFileName $3 $0\GvimExt32
 	  File /oname=$3 ${GETTEXT}\gettext32\libgcc_s_sjlj-1.dll
 	  Rename /REBOOTOK $3 $0\GvimExt32\libgcc_s_sjlj-1.dll
+  !endif
 !endif
 	${EndIf}
 	SetOverwrite lastused
@@ -452,7 +458,10 @@ Section "Native Language Support"
 	File ${GETTEXT}\gettext32\libintl-8.dll
 	File ${GETTEXT}\gettext32\libiconv-2.dll
 	#File /nonfatal ${VIMRT}\libwinpthread-1.dll
-	File /nonfatal ${GETTEXT}\gettext32\libgcc_s_sjlj-1.dll
+  !if /FileExists "${GETTEXT}\gettext32\libgcc_s_sjlj-1.dll"
+	# Install libgcc_s_sjlj-1.dll only if it is needed.
+	File ${GETTEXT}\gettext32\libgcc_s_sjlj-1.dll
+  !endif
 SectionEnd
 !endif
 
-- 
2.17.0

